app/adapters/whats_app_adapter/outbound.rb

```diff
@@ -4,94 +4,25 @@ module WhatsAppAdapter
-        WhatsAppAdapter::Outbound::Text.perform_later(recipient: contributor, text: text)
+        "WhatsAppAdapter::#{business_solution_provider}Outbound".constantize.send_more_info_message!(contributor)
       end
 
       def send_unsubsribed_successfully_message!(contributor)
-        return unless contributor_can_receive_messages?(contributor)
-
-        text = [I18n.t('adapter.whats_app.unsubscribe.successful'), "_#{I18n.t('adapter.whats_app.subscribe.instructions')}_"].join("\n\n")
-        WhatsAppAdapter::Outbound::Text.perform_later(recipient: contributor, text: text)
-      end
-
-      def contributor_can_receive_messages?(recipient)
-        recipient&.whats_app_phone_number.present?
-      end
-
-      def time_of_day
-        current_time = Time.current
-        morning = current_time.change(hour: 6)
-        day = current_time.change(hour: 11)
-        evening = current_time.change(hour: 17)
-        night = current_time.change(hour: 23)
-
-        case current_time
-        when morning..day
-          'morning'
-        when day..evening
-          'day'
-        when evening..night
-          'evening'
-        else
-          'night'
-        end
-      end
-
-      def freeform_message_permitted?(recipient)
-        responding_to_template_message = recipient.whats_app_message_template_responded_at.present? &&
-                                         recipient.whats_app_message_template_responded_at > 24.hours.ago
-        latest_message_received_within_last_24_hours = recipient.replies.first&.created_at.present? &&
-                                                       recipient.replies.first.created_at > 24.hours.ago
-        responding_to_template_message || latest_message_received_within_last_24_hours
-      end
-
-      def send_message_template(recipient, message)
-        recipient.update!(whats_app_message_template_sent_at: Time.current)
-        text = I18n.t("adapter.whats_app.request_template.new_request_#{time_of_day}_#{rand(1..3)}", first_name: recipient.first_name,
-                                                                                                     request_title: message.request.title)
-        WhatsAppAdapter::Outbound::Text.perform_later(recipient: recipient, text: text)
+        "WhatsAppAdapter::#{business_solution_provider}Outbound".constantize.send_unsubsribed_successfully_message!(contributor)
       end
 
-      def send_message(recipient, message)
-        files = message.files
+      private
 
-        if files.blank?
-          WhatsAppAdapter::Outbound::Text.perform_later(recipient: recipient, text: message.text)
-        else
-          files.each_with_index do |file, index|
-            WhatsAppAdapter::Outbound::File.perform_later(recipient: recipient, text: index.zero? ? message.text : '', file: file)
-          end
-        end
+      def business_solution_provider
+        Setting.three_sixty_dialog_client_api_key.present? ? 'ThreeSixtyDialog' : 'Twilio'
```

Comment:
Instead of returning a string here you could return a class

Reply:
I like this suggestion and have followed it in 176fbf31